context/User.tsx

```diff
@@ -0,0 +1,64 @@
+import React, {
+  useState,
+  useEffect,
+  createContext,
+  Dispatch,
+  SetStateAction,
+  ReactNode,
+} from "react";
+import { getSignInProfile } from "@/api/user";
+
+interface Props {
+  children: ReactNode;
+}
+
+interface IUserInfo {
+  token: string;
+  id: number;
+  name: string;
+  image_source: string;
+  email: string;
+}
+
+export const UserInfoContext = createContext<IUserInfo | null>(null);
+export const UserSetTokenContext = createContext<
+  Dispatch<SetStateAction<string | null>>
+>(() => {});
+
+function UserToken({ children }: Props) {
+  const [userInfo, setUserInfo] = useState<IUserInfo | null>(null);
+  const [userToken, setUserToken] = useState<string | null>(null);
+
+  const handleLoginUserInfo = async (token: string) => {
+    const result = await getSignInProfile(token);
+    const userInfo = result.data[0];
+
+    setUserInfo((prev) => ({
+      ...prev,
+      token: token,
+      id: userInfo.id,
+      name: userInfo.name,
+      image_source: userInfo.image_source,
+      email: userInfo.email,
```

Comment:
userInfo 에 id, name, image_source, email 만 포함되어있다면 아래처럼 작성할 수도 있겠네요 

```suggestion
      ...userInfo
```